http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcTest.java

```diff
@@ -161,6 +168,32 @@ public void testTokenSignatureAlgorithm() throws Exception {
         performAuthentication(getOidcConfigurationInputStreamWithTokenSignatureAlgorithm(), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
                 true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT);
     }
+    @Test
+    public void testInvalidScope() throws Exception {
+        checkInvalidScopeError(getoidcConfigurationInputStreamWithScope(CLIENT_SECRET, "INVALID_SCOPE"), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), "error=invalid_scope", OIDC_SCOPE + "+INVALID_SCOPE");
+    }
+
+    @Test
+    public void testEmptyScope() throws Exception {
+        performAuthentication(getoidcConfigurationInputStreamWithScope(CLIENT_SECRET, ""), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT);
+        checkScopeValuesInHTTPServerResponse(OIDC_SCOPE);
```

Comment:
It looks like the `checkScopeValuesInHTTPServerResponse` method currently duplicates things that are already being done in `performAuthentication` . 

To avoid the duplication, it would be good to add a new `performAuthentication` method that also takes an expected scope parameter. If this new parameter is not null then we can check the header from the response to verify that the scope was set correctly.

The existing `performAuthentication` method could then be updated to call the new one with the new parameter set to `false`.